src/web.ts

```diff
@@ -1,136 +1,170 @@
+        });
+      } catch (e) {
+        reject(e);
+      }
+    })
   }
 
-  async logoutMParticleUser(_call: any): Promise<any> {
-    const identityCallback = (result: any) => {
-      if (result.getUser()) {
-        console.log('logging out of mParticle', _call);
+  public logoutMParticleUser(_call?: any): Promise<IdentityResult> {
+    return new Promise((resolve, reject) => {
+      try {
+        this.mParticle.Identity.logout({} as any, (result) => {
+          resolve(result);
+        });
+      } catch (e) {
+        reject(e);
       }
-    };
-    return this.mParticle.Identity.logout({} as any, identityCallback);
+    });
   }
 
-  async registerMParticleUser(call: { email: string, customerId: string, userAttributes: any }): Promise<any> {
-    return mParticle.Identity.login(this.identityRequest(call.email, call.customerId), function (result: any) {
-      if (!result) return;
-      const currentUser = result.getUser();
-      for (const [key, value] of Object.entries(call.userAttributes)) {
-        if (key && value) currentUser.setUserAttribute(key, value);
-      }
+  public registerMParticleUser(call: { email: string, customerId?: string, userAttributes: any }): Promise<any> {
+    return new Promise((resolve, reject) => {
+      this.mParticle.Identity.login(this.identityRequest(call.email, call.customerId), (result) => {
+        if (!result) {
+          reject();
+        }
+        const currentUser = result.getUser();
+        for (const [key, value] of Object.entries(call.userAttributes)) {
+          if (key && value) currentUser.setUserAttribute(key, (value as any).toString());
+        }
+        resolve(result);
+      });
     });
   }
 
-  async logMParticleEvent(call: { eventName: string, eventType: any, eventProperties: any }): Promise<any> {
-    return this.mParticle.logEvent(call.eventName, call.eventType, call.eventProperties);
+  public logMParticleEvent(call: { eventName: string, eventType: any, eventProperties: any }): void {
+    this.mParticle.logEvent(call.eventName, call.eventType, call.eventProperties);
   }
 
-  async logMParticlePageView(call: { pageName: string, pageLink: string }): Promise<any> {
-    const attributes = {[ this.mparticleCapacitorConfiguration.eventAttributesMap.pageView.url]: call.pageLink };
-    return this.mParticle.logPageView(
+  public logMParticlePageView(call: { pageName: string, pageLink: string, overrides?: { attributeName: string }}): void {
+    let attributeName = "page";
+    if (call?.overrides?.attributeName) {
+      attributeName = call.overrides.attributeName;
+    }
+    const attributes = {[attributeName]: call.pageLink };
+    this.mParticle.logPageView(
       call.pageName,
       attributes, // pageLink comes in as window.location.toString()
       // call.googleAnalyticsValue // {"Google.Page": window.location.pathname.toString()} // if you're using Google Analytics to track page views
     );
   }
 
-  async getAllUserAttributes(): Promise<any> {
```

Comment:
@dghathway these return types were updated because we can't actually get any type of return from the mp sdk.  It's a fire and forget situation, and we can't reliably wait for the sdk calls to finish.

Same for all the other changes of `Promise<...>` to `void`.